src/pages/VerdictPage.tsx

```diff
@@ -111,19 +122,65 @@ export default function VerdictPage() {
             problemsObject.getProblem(data.problem_id)?.title ||
             'Custom Submission'
 
-          // Format the date string
-          const dateString = `${month}-${day}-${year}`
+          setProblemName(problemName)
+
+          let langStr = 'unknown'
+          if (data.language === 'cpp') {
+            langStr = 'C++'
+          } else if (data.language === 'java') {
+            langStr = 'Java'
+          } else if (data.language === 'py') {
+            langStr = 'Python'
+          }
+
+          setVerdictNum(data.verdict)
+          const verdictStr = isFinishedJudging
+            ? verdictInfo[data.verdict].description
+            : 'Pending'
+
+          const timeSeconds = data.time
+          const timeMilliseconds = Math.ceil(timeSeconds * 1000)
+          const timeStr = timeMilliseconds + ' ms'
+
+          const memoryKilobytes = data.memory
+          const memoryMegaBytes = Math.ceil(memoryKilobytes / 1000)
```

Comment:
I'm wondering if converting it to MB is the right move. Codeforces is using KB and I do see some of my submissions taking less than a megabyte (for example, 400 KB, 800 KB), although not many. In these cases, we would lose some info.

Reply:
I'm open to discussing this further. Here's why I think MB is better:
I think the only time its important to see your memory consumption is when you're concerned it might be large. It might be difficult to read something like 1035221 KB and know that it's actually less than 1GB. I agree that some precision is lost with MB though. Feel free to convince me otherwise. I'm curious what Tyler thinks.

Reply:
I see your point. What about using MB if it is at least 1000 KB? Otherwise, use KB.